lib/2wp-utils.js

```diff
@@ -245,7 +248,17 @@ const createSenderRecipientInfo = async (rskTxHelper, btcTxHelper, type = 'legac
     await rskTxHelper.importAccount(rskRecipientRskAddressInfo.privateKey);
     await rskTxHelper.unlockAccount(rskRecipientRskAddressInfo.address);
     if(Number(initialAmountToFundInBtc) > 0) {
-        await btcTxHelper.fundAddress(btcSenderAddressInfo.address, initialAmountToFundInBtc);
+        const initialAddressBalance = await btcTxHelper.getAddressBalance(btcSenderAddressInfo.address);
+        const fundTxHash = await btcTxHelper.fundAddress(btcSenderAddressInfo.address, initialAmountToFundInBtc);
+        const finalAddressBalance = await btcTxHelper.getAddressBalance(btcSenderAddressInfo.address);
+        // If the final address balance is the same as the initial balance, then the tx has not yet reached the mempool or has not been mined. Rare condition that may happen randomly.
+        if(finalAddressBalance === initialAddressBalance) {
+            const inMempool = await waitForBitcoinTxToBeInMempool(btcTxHelper, fundTxHash);
+            if(inMempool) {
+                await btcTxHelper.mine();
```

Comment:
What happens if we do this always? Instead of only doing it when initialBalance == finalBalance?

No big deal but I feel it would simplify the method